integration-tests/test_satellite.py

```diff
@@ -0,0 +1,108 @@
+import pytest
+import subprocess
+import os
+import yaml
+import traceback
+import contextlib
+from time import sleep
+import re
+
+
+pytestmark = pytest.mark.usefixtures("register_subman")
+
+
+def test_obfuscation(insights_client, tmp_path):
+    """
+    Test if the satellite hostname is obfuscated
+    Related to https://bugzilla.redhat.com/show_bug.cgi?id=1963917
+
+    """
+    # Register system to insights
+    insights_client.register()
+    # assert insights_client.is_registered
+
+    # Record the current hostname
+    with open("/etc/hostname", "r") as f:
+        previous_hostname = f.read().strip()
+
+    try:
+        # Set a new hostname
+        hostname_to_test = "insights-client-satellite"
+        set_hostname(hostname_to_test)
+        cmd = ["hostnamectl", "status"]
+        output = subprocess.check_output(cmd, universal_newlines=True)
+        assert hostname_to_test in output
+
+        # redaction on hostname:
+        data_redaction(hostname_to_test)
+
+        # check the collection, if redaction pattern is filtered out
+        # collect data
+        archive_result = insights_client.run("--no-upload")
+
+        # extract the collected tar file
+        m = re.search("^Archive saved at (.+)$", archive_result.stdout, re.MULTILINE)
+        assert m
+        archive_location = m.group(1)
+        cmd = "tar -xzvf %s -C %s" % (archive_location, tmp_path)
+        subprocess.check_call(cmd, shell=True)
```

Comment:
i'd split this command invocation into a list, passing that to `subprocess.check_call()` with no need for `shell=True`; using the shell as less as possible
1. makes command executions faster
2. avoids the need to deal with shell quoting, which is always a potential recipe for troubles